src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWebCSharp.1/Components/Account/IdentityComponentsEndpointRouteBuilderExtensions.cs

```diff
@@ -62,24 +66,27 @@ public static IEndpointConventionBuilder MapAdditionalIdentityEndpoints(this IEn
 
             var userId = await userManager.GetUserIdAsync(user);
             var userName = await userManager.GetUserNameAsync(user) ?? "User";
-            var userEntity = new PasskeyUserEntity(userId, userName, displayName: userName);
-            var passkeyCreationArgs = new PasskeyCreationArgs(userEntity);
-            var options = await signInManager.ConfigurePasskeyCreationOptionsAsync(passkeyCreationArgs);
-            return TypedResults.Content(options.AsJson(), contentType: "application/json");
+            var optionsJson = await signInManager.MakePasskeyCreationOptionsAsync(new()
+            {
+                Id = userId,
+                Name = userName,
+                DisplayName = userName
+            });
+            return TypedResults.Content(optionsJson, contentType: "application/json");
         });
 
         accountGroup.MapPost("/PasskeyRequestOptions", async (
+            HttpContext context,
             [FromServices] UserManager<ApplicationUser> userManager,
             [FromServices] SignInManager<ApplicationUser> signInManager,
+            [FromServices] IAntiforgery antiforgery,
             [FromQuery] string? username) =>
         {
+            await antiforgery.ValidateRequestAsync(context);
```

Comment:
Why not put the antiforgery metadata on the endpoint and have the app require the antiforgery middleware? (just a question)

Reply:
The app already does use antiforgery middleware, but it doesn't apply to the options endpoints because they're not form-bound. Adding the metadata manually would require adding a class that implements `IAntiforgeryMetadata`, right? Maybe there's a simpler way to do it that I'm not aware of 🙂